lib/elixir/lib/map_set.ex

```diff
@@ -63,6 +63,47 @@ defmodule MapSet do
   # with Elixir 1.14 and earlier and executed on Elixir 1.15+.
   defstruct map: :sets.new(version: 2)
 
+  @doc """
+  Checks if `map_set` contains `value`. Allowed in guards.
+
+  Otherwise functions like `member?/2`.
+
+  ## Examples
+
+      iex> MapSet.is_member(MapSet.new([1, 2, 3]), 2)
+      true
+
+      iex> MapSet.is_member(MapSet.new([1, 2, 3]), 4)
+      false
+
+      iex> case MapSet.new([1, 2, 3]) do
+      ...>   map_set when MapSet.is_member(map_set, 3) -> true
+      ...>   _ -> false
+      ...> end
+      true
+
+  """
+  @doc guard: true, since: "1.17.0"
+  defmacro is_member(map_set, value) do
+    case Macro.Env.in_guard?(__CALLER__) do
+      true ->
+        quote do
+          (unquote(map_set).__struct__ == unquote(__MODULE__) or :fail) and
+            is_map_key(unquote(map_set).map, unquote(value))
+        end
+
+      false ->
+        quote do
+          value = unquote(value)
+
+          case unquote(map_set) do
+            %MapSet{map: map} -> is_map_key(map, value)
+            other -> raise ArgumentError, "expected a MapSet, got: #{inspect(other)}"
+          end
```

Comment:
Maybe it should be:

```suggestion
          MapSet.member?(unquote(map_set), unquote(value))
```

Reply:
Agreed, much easier to keep in sync this way